app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.java

```diff
@@ -0,0 +1,279 @@
+            || p.state().equals(Project.STATE_CANCELED)
+            || p.state().equals(Project.STATE_FAILED)
+            || p.state().equals(Project.STATE_SUSPENDED)
+        );
+
+      this.projectOutput = this.project;
+
+      this.backersCountText = this.project
+        .map(Project::backersCount)
+        .map(NumberUtils::format);
+
+      this.backingViewGroupIsGone = this.project
+        .map(p -> p.isBacking() || p.isStarred() || p.isPotdToday() || p.isFeaturedToday());
+
+      this.blurbText = this.project
+        .map(Project::blurb);
+
+      this.categoryNameText = this.project
+        .map(Project::category)
+        .map(Category::name)
+        .compose(coalesce(""));
+
+      this.deadlineCountdownText = this.project
+        .map(ProjectUtils::deadlineCountdownValue)
+        .map(NumberUtils::format);
+
+      this.featuredViewGroupIsGone = this.project
+        .map(p -> p.isBacking() || p.isStarred() || p.isPotdToday() || !p.isFeaturedToday());
+
+      this.friendAvatarUrl = this.project
+        .filter(Project::isFriendBacking)
+        .map(Project::friends)
+        .map(friends -> friends.get(0).avatar().small());
+
+      this.friendBackingViewIsHidden = this.project
+        .map(Project::isFriendBacking)
+        .map(BooleanUtils::negate);
+
+      this.friendsForNamepile = this.project
+        .filter(Project::isFriendBacking)
+        .map(Project::friends);
+
+      this.fundingUnsuccessfulTextViewIsGone = this.project
+        .map(p ->
+          !p.state().equals(Project.STATE_CANCELED) &&
+            !p.state().equals(Project.STATE_FAILED) &&
+            !p.state().equals(Project.STATE_SUSPENDED));
+
+      this.imageIsInvisible = this.project
+        .map(Project::photo)
+        .map(ObjectUtils::isNull);
+
+      this.metadataViewGroupIsGone = this.project
+        .map(p -> !p.isBacking() && !p.isStarred() && !p.isPotdToday() && !p.isFeaturedToday());
+
+      this.nameText = this.project
+        .map(Project::name);
+
+      this.notifyDelegateOfProjectClick = this.project
+        .compose(Transformers.takeWhen(this.projectClicked));
+
+      this.percentageFunded = this.project
+        .map(Project::percentageFunded)
+        .map(ProgressBarUtils::progress);
+
+      this.percentageFundedProgressBarIsGone = projectInEndedStates;
+
+      this.percentageFundedText = this.project
+        .map(Project::percentageFunded)
+        .map(NumberUtils::flooredPercentage);
+
+      this.photoUrl = this.project
+        .map(Project::photo)
+        .filter(ObjectUtils::isNotNull)
+        .map(Photo::med);
+
+      this.potdViewGroupIsGone = this.project
+        .map(p -> p.isBacking() || p.isStarred() || !p.isPotdToday() || p.isFeaturedToday());
+
+      this.projectCanceledAt = this.project
+        .filter(p -> p.state().equals(Project.STATE_CANCELED))
+        .map(Project::stateChangedAt)
+        .compose(coalesce(new DateTime()));
```

Comment:
im a lil worried with these separate canceled/failed/successful/suspended/etc outputs. the main worry is that we are filtering on them, which means that as view holders are reused there is a chance that we don't reset their state cause the project was filtered out, and that could lead to a card rendering with two or more states :/

if you look at the screenshots for the 4 states we are concerned with:

[canceled](https://github.com/kickstarter/ios-oss/blob/master/Screenshots/_64/Kickstarter_Framework_iOSTests.DiscoveryPageViewControllerTests/testView_Card_Project_state_canceled_lang_en_device_phone4_7inch%402x.png) • [suspended](https://github.com/kickstarter/ios-oss/blob/master/Screenshots/_64/Kickstarter_Framework_iOSTests.DiscoveryPageViewControllerTests/testView_Card_Project_state_suspended_lang_en_device_phone4_7inch%402x.png) • [failed](https://github.com/kickstarter/ios-oss/blob/master/Screenshots/_64/Kickstarter_Framework_iOSTests.DiscoveryPageViewControllerTests/testView_Card_Project_state_failed_lang_en_device_phone4_7inch%402x.png) • [successful](https://github.com/kickstarter/ios-oss/blob/master/Screenshots/_64/Kickstarter_Framework_iOSTests.DiscoveryPageViewControllerTests/testView_Card_Project_state_successful_lang_en_device_phone4_7inch%402x.png)

you'll see that it mostly boils down to just a label for the state and a label for the date. perhaps we can merge all of that into just one observable for each label?